src/mesh/index_derivs.cxx

```diff
@@ -1067,23 +1067,103 @@ const Field2D Mesh::applyXdiff(const Field2D &var, Mesh::deriv_func func, Mesh::
 }
 
 const Field3D Mesh::applyXdiff(const Field3D &var, Mesh::deriv_func func, Mesh::inner_boundary_deriv_func func_in, Mesh::outer_boundary_deriv_func func_out, CELL_LOC loc) {
-  if (var.getNx() == 1){
+  if (var.getNx() == 1) {
     return 0.;
   }
+  
+  // Check that the input variable has data
+  ASSERT1(var.isAllocated());
+
   Field3D result;
   result.allocate(); // Make sure data allocated
+  
+  if (mesh->StaggerGrids && 
+      (loc != CELL_DEFAULT) && (loc != var.getLocation())) {
+    // Staggered differencing
 
-  bindex bx;
+    CELL_LOC location = var.getLocation();
+    
+    if (mesh->xstart > 1) {
+      // More than one guard cell, so set pp and mm values
+      // This allows higher-order methods to be used
+      for(const auto &i : result.region(RGN_NOX)) {
```

Comment:
I think this was to keep the same behavior as the old code. Agree it is inconsistent, but I think it's for taking mixed X-Z derivatives without having to communicate. Perhaps the region should be an optional input?